src/auth/auth-manager.js

```diff
@@ -0,0 +1,168 @@
+const jwt = require('jsonwebtoken');
+const User = require('../users/user-models').User;
+const Session = require('./auth-models').Session;
+
+/**
+ * Callback for sending the response to the client.
+ *
+ * @function loginResponse
+ * @param {(string|Object)} err - The error.
+ * @param {string} token - The JWT token.
+ */
+
+/**
+ * Given a valid email/password, generates and returns a JWT.
+ * @param {string} email - The user's email address.
+ * @param {string} password - The user's password.
+ * @param {loginResponse} next - The callback function to run after this function
+ *     finishes.
+ */
+const login = (email, password, next) => {
+  User.findOne({ email: email }, (err, user) => {
+    if (err) {
+      next(err);
+    } else if (!user) {
+      next('Auth failed');
+    } else {
+      user.comparePassword(password, (err, isMatch) => {
+        if (err) {
+          next(err);
+        } else {
+          Session.findOne({ email: email }, (err, session) => {
+            if (err) {
```

Comment:
Why this else statement? Would `next(err, session.token)` complete the same task?